scss/typography/_base.scss

```diff
@@ -441,3 +433,26 @@ $abbr-underline: 1px dotted $black !default;
     }
   }
 }
+
+/// Calculates the header font-size based on breakpoints.
+///
+/// @param {String} the size of the header to output.
+@mixin typography-header-font-size (
+  $requested-size
+) {
+  // Heading sizes
+  @each $size, $headers in $header-sizes {
+    @include breakpoint($size) {
```

Comment:
I agree that this would serve better as two separate mixins. Especially for things that are not included in the final output css, I far prefer explicit to brief.